### ISSUE TITLE
[BUG]: DIfferent error in compare_models() that in the predict function with same model and same data

### ISSUE THREAD

**Issue by alejandrogomez97** [pycaret/pycaret#3263]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

Hi.

I'm having a problem with the compare_models() function. When i run the compare_models() function it shows a mae of the first model of 263.7408. However, when i run the next line I obtain a MAE of 272.3426. This happens with all the models. Apparently, in the compare_models function they give a smaller error than the true error. How is this possible?

This is the code i'm using:

puntos=10000
inicio=0
train=df[inicio:-puntos]
test=df[-puntos:]

s = setup(data = train, target = 'lifetime', test_data=test, ignore_features=['duracion'],
fold_strategy = 'timeseries',
fold = 10,silent=True)

best3=compare_models(n_select=3)

predictions = predict_model(best3[0], data=test)

### Reproducible Example

```python
s = setup(data = train, target = 'lifetime', test_data=test, ignore_features=['duracion'],
fold_strategy = 'timeseries',
fold = 10,silent=True)

best3=compare_models(n_select=3)

predictions = predict_model(best3[0], data=test)
```

### Expected Behavior

I would expect the same error in both cases, as the data and the model are identical.

### Actual Results

```python-traceback
I get different errors. Usually, the error in the compare_models is smaller.
```

### Installed Versions

<details>
Replace this line with the output of the version code above.
</details>

**Comment by ngupta23** [comment]
The error metrics that you see in compare_models are for the cross validation. The error that you see in predict_model is for the unseen test data that you are passing. They are expected to be different. 

The error in compare_models being smaller than predict_model would imply overfitting. The extent of overfitting would need to be evaluated by the user to determine if it is acceptable.

**Comment by alejandrogomez97** [comment]
Isn't there a way to see in the compare_models the error on the test data? Apparently, instead of getting the best models for my ML problem, i'm getting the models with higher tendency to overfit my data.

**Comment by ngupta23** [comment]
We should not use the unseen "test" dataset to make decisions on model comparison. This should only be based on the CV results.

**Comment by alejandrogomez97** [comment]
Which is the set where we apply the CV? Is it part of the training set?

**Comment by ngupta23** [comment]
CV is already applied as part of compare_models. The results that you see from compare_models is the average CV metric across all folds.

**Comment by alejandrogomez97** [comment]
If we're applying timeseries k fold CV as in the scheme below, I don't understand how is it possible that the CV error is always smaller than the error in the test set. The CV error is based on the respective test set of each fold. This set isn't used for training the model in the corresponding step, so this way I should obtain a very representative error of the error in my real test set. How is it possible that I always obtain a smaller error in CV?

Thanks for your attention, Mr Gupta.

![image](https://user-images.githubusercontent.com/116890811/214297841-48be6f19-e89d-4bb5-a85f-c8754f50ea5d.png)

**Comment by ngupta23** [comment]
The terminology is different. What is called Test set in the image you posted is the CV holdout in pycaret. Test set is completely unseen. It is basically data that comes after the green bar to the right (assuming time is flowing from left to right).

**Comment by ngupta23** [comment]
For clarity, have a look at the time series notebooks (time series modules) in the example folder. There are a couple of plots that show the train-test split and how the train set is divided into CV folds.

**Comment by alejandrogomez97** [comment]
Ok thank you.
